Sources/Rendering/Core/ImageCPRMapper/index.js

```diff
@@ -0,0 +1,350 @@
+      if (nPoints <= 1) {
+        // Empty centerline
+        model._orientedCenterline = vtkPolyLine.newInstance();
+        return model._orientedCenterline;
+      }
+      // Copy the points to a new data array
+      const pointIndices = linesData.subarray(1, 1 + nPoints);
+      const centerlinePoints = new Float32Array(3 * nPoints);
+      const tempPoint = new Float32Array(3);
+      for (let i = 0; i < nPoints; ++i) {
+        const pointIdx = pointIndices[i];
+        pointsDataArray.getPoint(pointIdx, tempPoint);
+        centerlinePoints.set(tempPoint, 3 * i);
+      }
+      const points = vtkPoints.newInstance({
+        name: 'Points',
+        numberOfComponents: 3,
+        values: centerlinePoints,
+      });
+
+      // Get orientations of the centerline
+      const orientations = new Array(nPoints);
+      // Function to convert from mat4, mat3, quat or vec3 to quaternion
+      let convert = () => null;
+      const numComps = model.useUniformOrientation
+        ? model.uniformOrientation.length
+        : orientationDataArray.getNumberOfComponents();
+      switch (numComps) {
+        case 16:
+          convert = mat4.getRotation;
+          break;
+        case 9:
+          convert = (outQuat, inMat) => {
+            quat.fromMat3(outQuat, inMat);
+            quat.normalize(outQuat, outQuat);
+          };
+          break;
+        case 4:
+          convert = quat.copy;
+          break;
+        case 3:
+          convert = (a, b) => quat.rotationTo(a, model.tangentDirection, b);
+          break;
+        default:
+          vtkErrorMacro('Orientation doesnt match mat4, mat3, quat or vec3');
+          model._orientedCenterline = vtkPolyLine.newInstance();
+          return model._orientedCenterline;
+      }
+      // Function to get orientation from point index
+      let getOrientation = () => null;
+      if (model.useUniformOrientation) {
+        const outQuat = new Float64Array(4);
+        convert(outQuat, model.uniformOrientation);
+        getOrientation = () => outQuat;
+      } else {
+        const temp = new Float64Array(16);
+        getOrientation = (i) => {
+          const outQuat = new Float64Array(4);
+          orientationDataArray.getTuple(i, temp);
+          convert(outQuat, temp);
+          return outQuat;
+        };
+      }
+      // Fill the orientation array
+      for (let i = 0; i < nPoints; ++i) {
+        const pointIdx = pointIndices[i];
+        orientations[i] = getOrientation(pointIdx);
+      }
+
+      // Done recomputing
+      model._orientedCenterline = vtkPolyLine.newInstance({
+        points,
+        orientations,
+      });
+      return model._orientedCenterline;
+    }
+
+    // No polydata: return previous centerline
+    return model._orientedCenterline;
+  };
+
+  publicAPI.setOrientedCenterline = (centerline) => {
+    model._orientedCenterline = centerline;
+    // To trigger rebuild of distances
+    if (centerline) {
+      centerline.modified();
```

Comment:
does not seem right. 
If the centerline has not change, then it should not rebuild distances.